src/data/sym/basic.lean

```diff
@@ -313,3 +313,68 @@ multiset.mem_attach _ _
 coe_injective $ multiset.attach_cons _ _
 
 end sym
+
+section equiv
+
+/-! ### Combinatorial equivalences -/
+
+variables {α : Type*} {n : ℕ}
+open sym
+
+namespace sym_option_succ_equiv
+
+/-- Function from the symmetric product over `option` splitting on whether or not
+it contains a `none`. -/
+def encode [decidable_eq α] (s : sym (option α) n.succ) : sym (option α) n ⊕ sym α n.succ :=
+if h : none ∈ s
+then sum.inl (s.erase none h)
+else sum.inr (s.attach.map $ λ o,
+  option.get $ option.ne_none_iff_is_some.1 $ ne_of_mem_of_not_mem o.2 h)
+
+@[simp] lemma encode_of_none_mem [decidable_eq α] (s : sym (option α) n.succ) (h : none ∈ s) :
+  encode s = sum.inl (s.erase none h) := dif_pos h
+
+@[simp] lemma encode_of_not_none_mem [decidable_eq α] (s : sym (option α) n.succ) (h : ¬ none ∈ s) :
+  encode s = sum.inr (s.attach.map $ λ o,
+    option.get $ option.ne_none_iff_is_some.1 $ ne_of_mem_of_not_mem o.2 h) := dif_neg h
+
+/-- Inverse of `sym_option_succ_equiv.decode`. -/
+@[simp] def decode : sym (option α) n ⊕ sym α n.succ → sym (option α) n.succ
```

Comment:
Please use `sum.elim`.

Reply:
@urkud I'm not sure I understand the motivation, since this is readable and short as it is.

Reply:
Sorry, I forgot to delegate.
bors d+